api_client/python/timesketch_api_client/search.py

```diff
@@ -944,6 +977,40 @@ def save(self):
         self._resource_id = search_dict.get('id', 0)
         return f'Saved search to ID: {self._resource_id}'
 
+    def save_as_template(self):
+        """Save the search as a search template.
+
+        Returns:
+            A string indicating the search template ID.
+
+        Raises:
+            ValueError: if the search hasn't been saved first.
+            RuntimeError: If the search could not be saved as a template.
+        """
+        if not self._resource_id:
```

Comment:
the user is clearly want to save the search here, and then also create a template based on it. Maybe you can just save the search for the user if it isn't already? Or do you prefer to raise here and let the user do the needful?

Reply:
done